core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -222,6 +222,7 @@ private[spark] object ApiRootResource {
 private[spark] trait UIRoot {
   def getSparkUI(appKey: String): Option[SparkUI]
   def getApplicationInfoList: Iterator[ApplicationInfo]
+  def getApplicationInfoList(limit: Int): Iterator[ApplicationInfo]
```

Comment:
Ok somewhere in my testing I made a misassumption, but having the default be MaxInt would work fine too